app/controllers/users/sessions_controller.rb

```diff
@@ -2,9 +2,16 @@ class Users::SessionsController < Devise::SessionsController
   # before_filter :configure_sign_in_params, only: [:create]
 
   # GET /resource/sign_in
-  # def new
-  #   super
-  # end
+  def new
+    # set @oauth_authorize to disable sign-ins
+    # with third party applications (e.g. LinkedIn)
+    @oauth_authorize = false
+    if session['oauth_authorize'] == true
+      @oauth_authorize = true
+      session['oauth_authorize'] = false
```

Comment:
Why is this toggle neccesary? Because now, if user lands onto the log in page from the authorize page due to not being logged in, and inputs an invalid credentials, the LinkedIn button appears, which is not a good behavior. I think LinkedIn button should be hidden until user successfully logs in, or cancels the action and closes the browser window.